streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -192,17 +224,15 @@ public KafkaStreams(final TopologyBuilder builder, final StreamsConfig config, f
     public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
-        if (state == CREATED) {
+        if (state == KafkaStreamsState.CREATED) {
             for (final StreamThread thread : threads)
                 thread.start();
 
-            state = RUNNING;
+            state = KafkaStreamsState.RUNNING;
 
             log.info("Started Kafka Stream process");
-        } else if (state == RUNNING) {
-            throw new IllegalStateException("This process was already started.");
         } else {
-            throw new IllegalStateException("Cannot restart after closing.");
+            throw new IllegalStateException("Cannot start again.");
```

Comment:
Why change this? IMHO, the original message is more descriptive.

Furthermore, this exception should only occur for state "Pending Shutdown" and "Not Running". Right now, it would also fire for "Rebalancing"